core/src/main/java/com/homeaway/streamplatform/streamregistry/resource/ClusterResource.java

```diff
@@ -0,0 +1,177 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.homeaway.streamplatform.streamregistry.resource;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import lombok.extern.slf4j.Slf4j;
+
+import com.codahale.metrics.annotation.Timed;
+
+import io.dropwizard.jersey.errors.ErrorMessage;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import com.homeaway.streamplatform.streamregistry.db.dao.ClusterDao;
+import com.homeaway.streamplatform.streamregistry.model.ClusterKey;
+import com.homeaway.streamplatform.streamregistry.model.ClusterValue;
+import com.homeaway.streamplatform.streamregistry.model.JsonCluster;
+import com.homeaway.streamplatform.streamregistry.utils.ResourceUtils;
+
+@Api(value = "Stream-registry API", description = "Stream Registry API, a centralized governance tool for managing streams.")
+@Path("/v0/clusters")
+@Produces(MediaType.APPLICATION_JSON)
+@Slf4j
+public class ClusterResource {
+    private final ClusterDao clusterDao;
+
+    public ClusterResource(ClusterDao clusterDao) {
+        this.clusterDao = clusterDao;
+    }
+
+    @PUT
+    @ApiOperation(
+        value = "Upsert Clusters",
+        notes = "Create/Update a cluster",
+        tags = "clusters")
+    @ApiResponses(value = { @ApiResponse(code = 202, message = "Request accepted"),
+        @ApiResponse(code = 400, message = "Validation Exception while creating a cluster"),
+        @ApiResponse(code = 500, message = "Error Occurred while getting data") })
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/")
+    @Timed
+    public Response upsertCluster(@ApiParam(value = "Cluster Object", name = "Cluster Object", required = true) JsonCluster clusterParam) {
+        try {
+            log.info("Cluster Object {}", clusterParam);
+
+            clusterDao.upsertCluster(clusterParam);
+            return Response.status(Response.Status.ACCEPTED).build();
+        } catch (RuntimeException e) {
+            log.error("Error upserting cluster.", e);
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(new ErrorMessage(Response.Status.BAD_REQUEST.getStatusCode(),
+                    "Error upserting cluster.",
+                    e.getCause() != null ? e.getMessage() + ". " + e.getCause().getMessage() : e.getMessage()))
```

Comment:
can we add atleast the key for the error message
`"Error upserting cluster."`

Reply:
Added more details in the message when possible.

Reply:
i may put 
```
new ErrorMessage(Response.Status.BAD_REQUEST.getStatusCode(),
                    String.format(""Error upserting cluster:%s. %s", clusterParam, e.getMessage()),
                    e.getCause() != null ? e.getCause().getMessage() : null))
```